database/src/db.py

```diff
@@ -1,35 +1,123 @@
+                self.data = json.load(file)
+            print(f"Loaded {len(self.data)} items from data/cleaned_output.json")
+        except FileNotFoundError:
+            print("Warning: data/cleaned_output.json not found")
+            self.data = []
+
     def describe_index(self):
-        return self.index.describe_index_stats()
+        """Get statistics about the current index"""
+        if hasattr(self, 'index'):
+            return self.index.describe_index_stats()
+        else:
+            print(f"Index '{self.index_name}' is not available")
+            return None
+
+    def create_index(self, dimension=1536):
+        """Create a new Pinecone index with the given dimension"""
+        if self.pc.has_index(self.index_name):
+            print(f"Index '{self.index_name}' already exists")
+            return False
+        
+        self.pc.create_index(
+            name=self.index_name,
+            dimension=dimension,  # Default 1536 for OpenAI Ada-002
+            metric="cosine",
+            spec=ServerlessSpec(
+                cloud="aws",
+                region="us-east-1"
+            )
+        )
+        
+        print(f"Creating index '{self.index_name}'...")
+        
+        # Wait for index to be ready
+        import time
+        max_wait_time = 60  # seconds
+        start_time = time.time()
+        
+        while time.time() - start_time < max_wait_time:
+            try:
+                if self.pc.describe_index(self.index_name).status["ready"]:
+                    break
+                print("Waiting for index to be ready...")
+                time.sleep(5)
+            except Exception:
+                # Index might not be immediately available in describe call
+                time.sleep(5)
+        
+        print(f"Index '{self.index_name}' is ready")
+        
+        # Initialize the index object
+        self.index = self.pc.Index(self.index_name)
+        return True
+
+    def delete_all_vectors(self):
+        """Delete all vectors from the index"""
+        if not hasattr(self, 'index'):
+            print(f"Index '{self.index_name}' is not available")
+            return False
+            
+        try:
+            self.index.delete(delete_all=True, namespace="ns1")
+            print(f"All vectors deleted from index '{self.index_name}', namespace 'ns1'")
+            return True
+        except Exception as e:
+            print(f"Error deleting vectors: {e}")
+            return False
+
+    def delete_index(self):
+        """Delete the entire Pinecone index"""
+        if not self.pc.has_index(self.index_name):
+            print(f"Index '{self.index_name}' does not exist")
+            return False
+            
+        try:
+            self.pc.delete_index(self.index_name)
+            print(f"Index '{self.index_name}' deleted")
+            
+            # Remove the index attribute since it no longer exists
+            if hasattr(self, 'index'):
```

Comment:
related to a prev. comment, but i think it's better to keep the attribute defined so all the class variables are consistent?
but lmk!